packages/block-library/src/navigation-link/style.scss

```diff
@@ -61,6 +62,8 @@
 			}
 		}
 	}
+
+	// Show submenus on hover.
```

Comment:
Has the context for the decision to show these on hover changed? We could potentially remove the sensitive hover/focus-within rules entirely, and rely on &.is-selected or &.has-child-selected to keep it open. This would open the menu not just on hover, but on select too, and keep it open until deselected. This seems like it could simplify both the CSS and the user experience, and avoid the IE polyfill that is noted as necessary.

Reply:
I sm a fan of removing this on hover behavior and sticking to selection when editing. The only thing in my head to consider is that if the menu is indeed opening on hover when rendered, the user should be awayre that the edit behavior is different than the rendered behavior.

Reply:
Just in case there's any doubt, this PR doesn't change the behavior yet, just added a comment to describe the current behavior.

But yes, I think we can be opinionated about the edit behavior and keep it open on select. But we could _also_ show it on hover, and keep that behavior. And we'd still be able to remove the need for the suggested polyfill.

Once this PR lands, I'll try this separately, we can see how it feels 👍